From: Scott Talbert <scott.talbert@hgst.com>
Date: Mon, 9 May 2016 09:14:28 -0400
Patch-mainline: 4.7
Git-commit: 4673272f43ae790ab9ec04e38a7542f82bb8f020
References: bsc#1012452
Subject: [PATCH] btrfs: fix memory leak during RAID 5/6 device
 replacement

A 'struct bio' is allocated in scrub_missing_raid56_pages(), but it was never
freed anywhere.

Signed-off-by: Scott Talbert <scott.talbert@hgst.com>
Signed-off-by: David Sterba <dsterba@suse.com>
---
 fs/btrfs/scrub.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/fs/btrfs/scrub.c
+++ b/fs/btrfs/scrub.c
@@ -2129,6 +2129,8 @@ static void scrub_missing_raid56_end_io(
 	if (bio->bi_error)
 		sblock->no_io_error_seen = 0;
 
+	bio_put(bio);
+
 	btrfs_queue_work(fs_info->scrub_workers, &sblock->work);
 }
 
